python/pyarrow/_fs.pyx

```diff
@@ -395,6 +371,7 @@ cdef class FileSystem:
 
         stream.set_input_stream(in_handle)
         stream.is_readable = True
+        stream.is_seekable = True
```

Comment:
Hmm... I'm not sure why? If this is purely an `InputStream` then it's not seekable.
(and in practice e.g. a S3 input stream isn't seekable)

Reply:
I forgot to remove it, tried when I was debugging the `stream.read()` issue, which I still need to create a JIRA.